corehq/apps/locations/bulk_management.py

```diff
@@ -882,35 +881,18 @@ def append_at_bottom(parent_type):
 
         return top_to_bottom_locations
 
-    def _process_locations(locs_to_process, to_be_deleted):
-        for loc in locs_to_process:
+    delete_location_ids = []
+    with transaction.atomic():
+        for loc in order_by_location_type():
             if excel_importer:
                 excel_importer.add_progress()
             if loc.do_delete:
                 if not loc.is_new:
-                    # keep track of to be deleted items to delete them in top-to-bottom order
-                    to_be_deleted.append(loc.db_object)
+                    delete_location_ids.append(loc.location_id)
             elif loc.needs_save:
                 loc_object = loc.db_object
                 loc_object.location_type = types_by_code.get(loc.location_type)
                 loc_object.parent = loc.new_parent
-                loc.db_object.save()
-
-    to_be_deleted = []
-
-    top_to_bottom_locations = order_by_location_type()
-    if delay_updates:
-        for locs in chunked(top_to_bottom_locations, chunk_size):
-            with transaction.atomic():
-                with SQLLocation.objects.delay_mptt_updates():
-                    _process_locations(locs, to_be_deleted)
-    else:
-        _process_locations(top_to_bottom_locations, to_be_deleted)
-
-    for locs in chunked(reversed(to_be_deleted), chunk_size):
-        # Deletion has to happen bottom to top, otherwise mptt complains
-        #   about missing parents
-        with transaction.atomic():
-            with SQLLocation.objects.delay_mptt_updates():
-                for l in locs:
-                    l.delete()
+                loc_object.save()
+
+        SQLLocation.objects.filter(location_id__in=delete_location_ids).delete()
```

Comment:
well this got quite a bit cleaner. :sparkles:

Reply:
yeah, unfortunately this didn't work because there are other things associated with locations that live in other databases that also need to be cleaned up. ☹️